src/components/modals/AddUserModal.jsx

```diff
@@ -0,0 +1,240 @@
+import _ from 'lodash';
+import React, { Fragment, useState, useRef, useEffect } from 'react';
+import { User } from '../../libs/ajax';
+import { USER_ROLES } from '../../libs/utils';
+import { Alert } from '../Alert';
+import { BaseModal } from '../BaseModal';
+import addUserIcon from '../../images/icon_add_user.png';
+
+const adminRole = { 'roleId': 4, 'name': USER_ROLES.admin };
+const researcherRole = { 'roleId': 5, 'name': USER_ROLES.researcher };
+
+export const AddUserModal = (props) => {
+  const [state, setState] = useState({
+    displayName: '',
+    email: '',
+    displayNameValid: false,
+    emailValid: false,
+    invalidForm: true,
+    submitted: false,
+    alerts: [],
+    updatedRoles: [researcherRole],
+    emailPreference: false
+  });
+
+  const nameRef = useRef();
+  const emailRef = useRef();
+
+  useEffect(() => {
+    if (!props.showModal) return;
+    let r1 = nameRef.current;
+    let r2 = emailRef.current;
+    if (r1 && r2) {
+      setState((prev) => ({
+        ...prev,
+        displayNameValid: r1.validity.valid,
+        emailValid: r2.validity.valid
+      }));
+    }
+  }, [props.showModal]);
+
+  const OKHandler = async (event) => {
+    event.persist();
+    setState({
+      ...state,
+      submitted: true
+    });
+    const validForm = state.displayNameValid && state.emailNameValid;
```

Comment:
I think this was mis-transcribed:
```suggestion
    const validForm = state.displayNameValid && state.emailValid;
```